src/kernel/capinv.c

```diff
@@ -1294,7 +1294,7 @@ static int __attribute__((noinline)) composite_syscall_slowpath(struct pt_regs *
 			vaddr_t    entry_addr    = __userregs_get3(regs);
 			invtoken_t token         = __userregs_get4(regs);
 
-			ret = sinv_activate(ct, cap, capin, dest_comp_cap, entry_addr, token);
+			ret = sinv_activate(ct, cap, capin, dest_comp_cap, CAP_SINV, entry_addr, token);
```

Comment:
`sinv_activate` with a type? I think you should leave `sinv_activate` as is. In the FAULTACTIVATE, use `faultinv_activate` or something and call an internal function that would be `sinv_activate_type` within `inv.h`